aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PendingMutation.java

```diff
@@ -38,34 +37,55 @@
  * logic onto them, before ultimately uploading to a remote endpoint.
  * @param <T> Type of model that has experienced a mutation
  */
-public final class PendingMutation<T extends Model> {
+public final class PendingMutation<T extends Model> implements Comparable<PendingMutation<? extends Model>> {
     private final T mutatedItem;
     private final Class<T> classOfMutatedItem;
     private final Type mutationType;
+    private final TimeBasedUuid mutationId;
 
-    private PendingMutation(T mutatedItem, Class<T> classOfMutatedItem, Type mutationType) {
+    private PendingMutation(TimeBasedUuid mutationId, T mutatedItem, Class<T> classOfMutatedItem, Type mutationType) {
+        this.mutationId = mutationId;
         this.mutatedItem = mutatedItem;
         this.classOfMutatedItem = classOfMutatedItem;
         this.mutationType = mutationType;
     }
 
     /**
-     * Creates a {@link PendingMutation}.
+     * Creates a {@link PendingMutation}, using a provided mutation ID.
+     * @param mutationId A globally-unique, *time-based* ID for this mutation event;
+     *                   the ID is used for temporal ordering of mutations
      * @param mutatedItem The item that undergone a mutation
      * @param classOfMutatedItem The class of the item that has undergone mutation
      * @param mutationType Type of mutation
      * @param <T> The type of the item that has undergone mutation
      * @return A {@link PendingMutation}
      */
     static <T extends Model> PendingMutation<T> instance(
```

Comment:
Other than for testing purposes, do we want to leave this `instance` method callable from any class within this package? 

This is very low risk, but my only thought is that we want to ensure these time-based UUIDs are generated on the local device only right?

Reply:
This version is actually used by the Gson deserializer, when we convert back from a (peristable) `PersistentRecord`. This is how we "recover" the stored timestamp, back into a `PendingMutation`.